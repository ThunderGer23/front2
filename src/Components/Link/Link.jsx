import PropTypes from 'prop-types'

export const Link = ({texto}) =>{
    return(
        <>
        <div role='div' className="text-center pt-1 mb-1">
            <p role='p'><a role='a' className="">{texto}</a></p>
        </div>
        </>
    )
}

Link.propTypes = {
    texto: PropTypes.string.isRequired
}