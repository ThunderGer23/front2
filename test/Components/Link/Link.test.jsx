import { render, screen } from "@testing-library/react";
import { Link } from "../../../src/Components/Link/Link";

describe('Pruebas en el componente Link', () => { 
    const texto='y tu contraseÃ±a?'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<Link texto={texto}/>)//Renderiza el componente Link
        expect(container).toMatchSnapshot()//Toma foto
     })

    test('Esta prueba valida que el div exista y pertenezca a la clase correcta', () => { 
        render(<Link texto={texto}/>)
        expect(screen.findByRole('div')).toBeTruthy()
        const {className} = screen.getByRole('div')//Se salva el tipo de la clase del div en className
        expect(className).toBe('text-center pt-1 mb-1')
      })
      
    test('Esta prueba valida que el p exista', () => { 
        render(<Link texto={texto}/>)
        expect(screen.findByRole('p'))
     })

     test('Esta prueba valida que el a exista', () => { 
        render(<Link texto={texto}/>)
        expect(screen.findByRole('a'))
     })
 })