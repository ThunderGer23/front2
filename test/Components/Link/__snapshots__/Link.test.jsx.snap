// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Link Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class="text-center pt-1 mb-1"
    role="div"
  >
    <p
      role="p"
    >
      <a
        class=""
        role="a"
      >
        y tu contrase√±a?
      </a>
    </p>
  </div>
</div>
`;
